src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import functions.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    public static void main(String[] args) {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello Hello! I'm Duke\n" + "What can I do for you?");
+        int i = 0;
+        String txt;
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            i = DukeSave.load(tasks);
+        } catch (Exception m) {
+            System.out.println("An exception has occurred: " + m);
+        }
+
+        while (true) {
+            txt = in.nextLine();
+            try {
+                if (txt.equals("list")) {                        //CMD LIST
+                    System.out.println("Here are the tasks in your list:");
+                    for (int x = 0; x < i; x++) {
+                        System.out.printf("%d.%s\n", x + 1, tasks.get(x));
+                    }
+                } else if (txt.startsWith("done ")) {            //CMD DONE
+                    if (txt.equals("done ")) {
+                        throw new DukeException("No description entered ...");
```

Comment:
You can consider defining the Exception text as constants, since there are only a fixed number of errors for an application and their text is rarely changed during runtime.